### PR TITLE
[Security] Require builder access for datasource updates

### PR DESCRIPTION
**Pull request by PClmnt**
## Description
Move datasource update requests behind the existing builder-only datasource route group and add a regression test for the update endpoint authorization boundary.

## Addresses
- https://github.com/Budibase/vulns/issues/57
- https://github.com/Budibase/budibase/security/advisories/GHSA-44m2-crh7-f4q2

## App Export
- N/A

## Screenshots
- N/A, backend authorization fix only.

## Launchcontrol
Datasource updates now require builder access.

Tests run:
- `cd packages/server && CI=1 yarn test src/api/routes/tests/datasource.spec.ts -t "permissions"`

Also run:
- `cd packages/server && CI=1 yarn test src/api/routes/tests/datasource.spec.ts` (failed in unrelated integration-backed sections due local datasource containers: mssql container stopped, oracle connection reset, postgres timeout/400 setup failures; the new permissions test passed in that run).

---
## Summary by cubic
Require builder access for datasource updates to prevent unauthorized changes by production users. Moves the update endpoint into the builder-only route group.

- **Bug Fixes**
  - Moved `PUT /api/datasources/:datasourceId` to `builderRoutes`; removed from `authorizedRoutes`.
  - Added tests to verify builder-only access and that forbidden updates return 403 without persisting changes.

<sup>Written for commit 7746578103dab6859a88b06c73d1df223e17998a. Summary will update on new commits.</sup>

### CODE REVIEW

**Comment by PClmnt** [comment]
@cubic-dev-ai review

**Comment by PClmnt** [comment]
Tested locally and confirmed I can still update datasource names.

File: packages/server/src/api/routes/tests/datasource.spec.ts [hunk 1]
```diff
@@ -756,6 +756,15 @@ describe("/datasources", () => {
         url: `/api/datasources/${datasource._id}/${datasource._rev}`,
       })
     })
+
+    it("should apply authorization to update endpoint", async () => {
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/datasource.spec.ts:764]: Should be not test the non-authorized scenario?

**Review by adrinr — APPROVED** [review]

File: packages/server/src/api/routes/tests/datasource.spec.ts (hunk 1 above)
Comment (PClmnt) [reply packages/server/src/api/routes/tests/datasource.spec.ts:764]: Well I was affirming that it now does it, but yes, we should probably check the previous vulnerable scenario as well for a negative, good call

**Review by melohagan — APPROVED** [review]
